omnigan/data.py

```diff
@@ -22,8 +22,54 @@
     [".jpg", ".JPG", ".jpeg", ".JPEG", ".png", ".PNG", ".ppm", ".PPM", ".bmp", ".BMP"]
 )
 
+classes_dict = {
+    "s": {
+        0: [0, 0, 255, 255],  # Water
+        1: [55, 55, 55, 255],  # Ground
+        2: [0, 255, 255, 255],  # Building
+        3: [255, 212, 0, 255],  # Traffic items
+        4: [0, 255, 0, 255],  # Vegetation
+        5: [255, 97, 0, 255],  # Terrain
+        6: [255, 0, 0, 255],  # Car
+        7: [0, 0, 0, 0],  # Trees
+        8: [255, 0, 255, 255],  # Person
+        9: [0, 0, 0, 255],  # Sky
+        10: [255, 255, 255, 255],  # Default
+    },
+    "r": {
+        0: [0, 0, 255, 255],  # Water
+        1: [55, 55, 55, 255],  # Ground
+        2: [0, 255, 255, 255],  # Building
+        3: [255, 212, 0, 255],  # Traffic items
+        4: [0, 255, 0, 255],  # Vegetation
+        5: [255, 97, 0, 255],  # Terrain
+        6: [255, 0, 0, 255],  # Car
+        7: [0, 255, 0, 255],  # Trees
+        8: [220, 20, 60, 255],  # Person
+        9: [8, 19, 49, 255],  # Sky
+        10: [0, 80, 100, 255],  # Default
+    },
+}
+
+
+def decode_segmap_unity_labels(tensor, domain, is_target, nc=11):
+    """Creates a label colormap for classes used in Unity segmentation benchmark.
+    Arguments:
+        tensor -- segmented image of size (1) x (nc) x (H) x (W) if prediction, or size (1) x (1) x (H) x (W) if target
+    Returns:
+        RGB tensor of size (1) x (3) x (H) x (W)
+    # """
+
+    if is_target:  # Target is size 1 x 1 x H x W
+        idx = tensor.squeeze(0).squeeze(0)
+    else:  # Prediction is size 1 x nc x H x W
+        idx = torch.argmax(tensor.squeeze(0), dim=0)
+
+    indexer = torch.tensor(list(classes_dict[domain].values()))[:, :3]
```

Comment:
inspired by that question https://stackoverflow.com/questions/46868855/numpy-replacing-values-in-a-2d-array-efficiently-using-a-dictionary-as-a-map the implementation I pushed should be ~50-100 times faster

Reply:
Damn, cool! It was indeed slow haha. :)